File: reimplement/bluetooth-manager.lbaction/Contents/Scripts/choice.js
"use strict";

const lb = require("launchbar-node");
const { execFile } = require("@roeybiran/task");

const address = process.argv[2];

(async () => {
  try {
    lb.hide();
    const { stdout } = await execFile("/usr/local/bin/blueutil", [
      "--is-connected",
      address
    ]);
    let arg;
    if (stdout === "0") {
      arg = "--connect";
    } else if (stdout === "1") {
      arg = "--disconnect";
    }
    execFile("/usr/local/bin/blueutil", [arg, address]);
  } catch (error) {
    const msg =
      error.code === "ENOENT" ? "This action requires blueutil" : error;
    console.log(msg);
  }
})();

File: bluetooth-manager.lbaction/Contents/Scripts/choice.py
#!/usr/bin/env python

import sys
import plistlib
import subprocess
import json
import os
import re

address = sys.argv[1]
is_connected = subprocess.check_output(
    ["/usr/local/bin/blueutil", "--is-connected", address]) == "1"

if is_connected:
    argument = "--disconnect"
else:
    argument = "--connect"

subprocess.check_output(["/usr/local/bin/blueutil", argument, address])
